src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1886,17 +1886,37 @@ static bool processSubscriptions
     if (notificationSent)
     {
       long long rightNow = getCurrentTime();
-
+      BSONObj query  = BSON("_id" << OID(mapSubId));
+      BSONObj update;
       //
       // If broker running without subscription cache, put lastNotificationTime and count in DB
       //
       if (subCacheActive == false)
       {
-        BSONObj query  = BSON("_id" << OID(mapSubId));
-        BSONObj update = BSON("$set" <<
-                              BSON(CSUB_LASTNOTIFICATION << rightNow) <<
-                              "$inc" << BSON(CSUB_COUNT << (long long) 1));
+        BSONObj subOrig;
+        std::string newErr;
+        collectionFindOne(getSubscribeContextCollectionName(tenant), BSON("_id" << OID(mapSubId)), &subOrig, &newErr);
```

Comment:
You already have the query part in a variable. Why don't use?

```
collectionFindOne(getSubscribeContextCollectionName(tenant), query, &subOrig, &newErr);
```

Reply:
fixed in 00406ae